packages/plugin-commands-installation/src/link.ts

```diff
@@ -178,11 +183,14 @@ export async function handler (
       globalPkgNames = pkgNames
     }
     const globalPkgPath = pathAbsolute(opts.dir)
-    globalPkgNames.forEach((pkgName) => pkgPaths.push(path.join(globalPkgPath, 'node_modules', pkgName)))
+    globalPkgNames.forEach((pkgName) => {
+      const pkgPath = path.join(globalPkgPath, 'node_modules', pkgName)
+      pkgPaths.push(pkgPath)
+      const dependencyType = getDependencyTypeFromManifest(manifest, pkgName)
+      targetDependenciesFieldMap[pkgPath] = dependencyType ?? linkOpts.targetDependenciesField
```

Comment:
it'll be better if you pass the manifest to the `link()` function and get the depType inside the link function.

Reply:
That'll be better. Thanks